libethereum/BlockChain.cpp

```diff
@@ -805,26 +741,85 @@ ImportRoute BlockChain::import(VerifiedBlockRef const& _block, OverlayDB const&
 		auto tdIncrease = s.enactOn(_block, *this);
 
 		for (unsigned i = 0; i < s.pending().size(); ++i)
-		{
-			blb.blooms.push_back(s.receipt(i).bloom());
 			br.receipts.push_back(s.receipt(i));
-			goodTransactions.push_back(s.pending()[i]);
-		}
 
 		s.cleanup(true);
 
 		td = pd.totalDifficulty + tdIncrease;
 
-#if ETH_TIMED_IMPORTS
 		performanceLogger.onStageFinished("enactment");
-#endif // ETH_TIMED_IMPORTS
 
 #if ETH_PARANOIA
 		checkConsistency();
 #endif // ETH_PARANOIA
+	}
+#if ETH_CATCH
```

Comment:
Maybe it is safe to assume ETH_CATCH all the time. Having different implementations depending on ETH_CATCH macro seems to be bad idea (maybe it was useful in time of experimenting with it, but rather not useful now).